presto-thrift-connector/src/test/java/com/facebook/presto/connector/thrift/util/TestRetryDriver.java

```diff
@@ -85,7 +87,8 @@ void testRetrySuccess()
             throws Exception
     {
         SyncTaskImpl task = new SyncTaskImpl(SYNC_FAILURE, SYNC_FAILURE, SYNC_FAILURE, SYNC_FAILURE, SUCCESS);
-        Integer result = retry.run("test", task);
+        RetryStats retryStats = new RetryStats();
```

Comment:
I think there's value in adding a separate test which checks if stats are populated.

Reply:
Thanks! added test and found out a bug in counting async retry.